### DISCUSSION TITLE
Dialogic Animation

### DISCUSSION THREAD

**Discussion by AxzonZero** [dialogic-godot/dialogic#1600] [Q&A] [answered]
After my dialogic timeline ends, I want an animation to play. How do I do this? Someone told me to use **dialog.timeline_end** but I'm not sure what they mean by that.

**Comment by Jowan-Spooner** [comment, marked as answer]
Hi, it would be useful to know what version you are using.

Dialogic emits a signal when it's finished. If you are new to godot, try to learn about [signals here](https://docs.godotengine.org/en/stable/getting_started/step_by_step/signals.html).

In dialogic 1 the signal is part of the dialogic node returend by Dialogic.start(). This means something like this code should work:
```
    var dialog = Dialogic.start('my_timeline')
    add_child(dialog)
    dialog.connect('timeline_end', self, '_on_timeline_end')
    
func _on_timeline_end(timeline):
    print("The timeline ended")
    $AnimationPlayer.play('timeline_end_animation')
```

In dialogic 2 it could look like this:
```
    Dialogic.start('my_timeline')
    Dialogic.timeline_ended.connect(_on_timeline_end)
    
func _on_timeline_end():
    print("The timeline ended")
    $AnimationPlayer.play('timeline_end_animation')
```

I hope this helps.
